src/SW_Main_lib.c

```diff
@@ -191,18 +195,110 @@ void sw_init_args(int argc, char **argv, Bool *QuietMode,
 
 }
 
+/**
+ * @brief In SOILWAT2 and rSOILWAT2, if an error occurs, this function
+ *  notifies the user and crashes
+ *
+ * @param[in] QuietMode Flag to control if the program version is displayed
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void sw_check_exit(Bool QuietMode, LOG_INFO* LogInfo) {
+    #ifdef RSOILWAT
+    if(LogInfo->stopRun) {
+        error(LogInfo->errorMsg);
+    }
+
+    (void) QuietMode; // Silence compiler flag `-Wunused-parameter`
+    #else
+    if(LogInfo->stopRun) {
+        if(!QuietMode) {
+            fprintf(stderr, "%s", LogInfo->errorMsg);
+        }
+        exit(-1);
+    }
+    #endif
+}
 
-void sw_check_log(Bool QuietMode, LOG_INFO* LogInfo) {
-	/* =================================================== */
-	/* function to be called by atexit() so it's the last
-	 * to execute before termination.  This is the place to
-	 * do any cleanup or progress reporting.
-	 */
-	if (LogInfo->logfp != stdout && LogInfo->logfp != stderr) {
-		CloseFile(&LogInfo->logfp, LogInfo);
-		if (LogInfo->logged && !QuietMode) {
-			sw_error(0, "\nCheck logfile for error or status messages.\n");
-		}
+/**
+ * @brief Initialize values within LOG_INFO
+ *
+ * @param[in] logInitPtr Log file pointer that LOG_INFO will be initialized to
+ * @param[in,out] LogInfo Holds information dealing with log file output
+*/
+void sw_init_logs(FILE* logInitPtr, LOG_INFO* LogInfo) {
+
+	LogInfo->logged = swFALSE;
+	LogInfo->logfp = logInitPtr;

```

Comment:
Currently, we are not deleting/removing the logfile content from a previous SOILWAT2 run. For instance, we run SOILWAT2 and this leads to an error and/or warning, then that error and/or warning text is written to the logfile. We fix the inputs and now SOILWAT2 runs without errors and warnings, but the logfile still shows the error and/or warning from the previous run.

We need to reset/delete the content of the logfile, e.g., in `SW_F_read()` -- although, I thought that ``LogInfo->logfp = OpenFile(PathInfo->InFiles[eLog], "w", LogInfo);` should reset the logfile content based on https://en.cppreference.com/w/c/io/fopen : file access mode "w", then action if file already exists is "destroy content". Why doesn't it?

Reply:
Ah, my text editor didn't update cached content if content was deleted... all good